### PR TITLE
let mcmc run with empty latent sites

### PR DESCRIPTION
**Pull request by fehiepsi**
Address #351. This PR fixes a problem which happens in an empty model: step_size keeps increasing forever in `find_reasonable_step_size` function.

@neerajprad probably we don't want to run mcmc at all if the model has no latent site, but it is not addressed in this PR. I let you justify the correct behavior in that case.

### CODE REVIEW

File: numpyro/mcmc.py [hunk 1]
```diff
@@ -738,7 +738,7 @@ def _single_chain_mcmc(self, init, collect_fields=('z',), collect_warmup=False,
         if len(collect_fields) == 1:
             samples = (samples,)
         samples = dict(zip(collect_fields, samples))
-        samples['z'] = vmap(constrain_fn)(samples['z'])
+        samples['z'] = vmap(constrain_fn)(samples['z']) if len(tree_flatten(samples)[0]) > 0 else samples['z']
```
Comment (neerajprad) [inline numpyro/mcmc.py]: if `samples['z']` is empty does vmap complain?

**Comment by neerajprad** [comment]
> @neerajprad probably we don't want to run mcmc at all if the model has no latent site, but it is not addressed in this PR. I let you justify the correct behavior in that case.

I think it is better to just let it run through the code rather than short-circuiting it, since it should be quite fast anyhow.

File: numpyro/mcmc.py (hunk 1 above)
Comment (fehiepsi) [reply numpyro/mcmc.py]: Yes, `vmap` will complain about it: `import jax; jax.vmap(lambda x: x)({})` triggers
```
ValueError: not enough values to unpack (expected 1, got 0)
```

**Comment by fehiepsi** [comment]
Agreed! It is quite fast! :D

**Pull request merged**